### PR TITLE
React. Components

### PR DESCRIPTION
**Pull request by AleksGF**
1. [Task](https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module01#react-components)
2. Screenshot
screenshot

3. [Repository](https://github.com/AleksGF/react-course/tree/module1)
4. Done: 29.10.2023 / Deadline: 30.10.2023
5. Score: 100 / 100

### CODE REVIEW

File: .eslintcache
Comment (AnAtoliy-AA) [inline .eslintcache:1]: you can add it to .gitignore

File: .husky/pre-commit
Comment (AnAtoliy-AA) [inline .husky/pre-commit:1]: greate!

File: package.json [hunk 1]
```diff
@@ -20,9 +22,22 @@
     "@typescript-eslint/parser": "^6.0.0",
     "@vitejs/plugin-react": "^4.0.3",
     "eslint": "^8.45.0",
+    "eslint-config-prettier": "^9.0.0",
+    "eslint-plugin-prettier": "^5.0.1",
+    "eslint-plugin-react": "^7.33.2",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.4.3",
```
Comment (AnAtoliy-AA) [inline package.json:29]: https://www.knowledgehut.com/blog/web-development/npm-install-dev-dependencies

File: src/App.tsx [hunk 2]
```diff
@@ -1,35 +1,105 @@
...
+  state: AppState = {
+    shouldUpdateData: false,
+    isLoading: false,
+    currentSearch: null,
+    searchValue: '',
+    itemsToShow: [],
+  };
+
+  updateData(search) {
+    if (search === this.state.currentSearch) return;
```
Comment (AnAtoliy-AA) [inline src/App.tsx:39]: `shouldComponentUpdate` ?
https://www.geeksforgeeks.org/what-does-shouldcomponentupdate-do-and-why-is-it-important/

File: src/App.tsx [hunk 3]
```diff
@@ -1,35 +1,105 @@
...
+    });
+  }
+
+  componentDidMount() {
+    this.updateData(localStorage.getItem('rc_lastSearch') ?? '');
+  }
+
+  componentDidUpdate() {
+    if (this.state.shouldUpdateData) {
+      const searchValue = this.state.searchValue.trim();
```
Comment (AnAtoliy-AA) [inline src/App.tsx:63]: empty string

File: src/App.tsx [hunk 4]
```diff
@@ -1,35 +1,105 @@
...
+      }));
+    });
+  }
+
+  componentDidMount() {
+    this.updateData(localStorage.getItem('rc_lastSearch') ?? '');
+  }
+
+  componentDidUpdate() {
+    if (this.state.shouldUpdateData) {
```
Comment (AnAtoliy-AA) [inline src/App.tsx:62]: look at coment before)

File: src/components/ContentFrame/ContentFrame.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+          <h2>There are no persons to display</h2>
+        </div>
+      );
+
+    return (
+      <div className={'content-frame__wrapper'}>
+        <h2 className={'content-frame__title'}>Star War Persons</h2>
+        {this.props.people.map((person) => (
+          <ContentItem
+            key={`${person.name}-${person.url}-${person.created}`}
```
Comment (AnAtoliy-AA) [inline src/components/ContentFrame/ContentFrame.tsx:23]: good

File: src/components/ErrorBoundary/ErrorBoundary.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    hasError: false,
+  };
+
+  static getDerivedStateFromError() {
+    return { hasError: true };
+  }
+
+  componentDidCatch(error, info) {
+    console.log(error.message);
+    console.log(info.componentStack);
```
Comment (AnAtoliy-AA) [inline src/components/ErrorBoundary/ErrorBoundary.tsx:18]: lofg or maybe warn or maybe u don't need it

File: src/components/common/Button/Button.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { ButtonProps } from '../../../types/types';
+import './Button.scss';
+
+class Button extends React.Component<
+  React.PropsWithoutRef<ButtonProps>,
+  undefined
```
Comment (AnAtoliy-AA) [inline src/components/common/Button/Button.tsx:7]: why you use undefined here?

File: src/components/common/ComponentWithError/ComponentWithError.tsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+class ComponentWithError extends React.Component<undefined, undefined> {
```
Comment (AnAtoliy-AA) [inline src/components/common/ComponentWithError/ComponentWithError.tsx:3]: <undefined, undefined> ??

File: package.json (hunk 1 above)
Comment (AleksGF) [reply package.json:29]: Thank you very much!
I`ll try to correct all issues in current module!

**Pull request closed without merging**